src/toil/jobStores/aws/utils.py

```diff
@@ -193,78 +190,80 @@ def fileSizeAndTime(localFilePath):
     return file_stat.st_size, file_stat.st_mtime
 
 
-def uploadFromPath(localFilePath, partSize, bucket, fileID, headers):
+def uploadFromPath(localFilePath, resource, bucketName, fileID, headerArgs=None, partSize=50 << 20):
```

Comment:
Including this typing in the args would be awesome.  For example:

```
from typing import Optional  # place at top of file

uploadFile(localFilePath: str, resource, bucketName: str, fileID: str, headerArgs: Optional[dict] = None, partSize: int = 50 << 20):
```

The objects are a little trickier and would have to be hunted down.